optuna/integration/lightgbm_tuner/sklearn.py

```diff
@@ -1,58 +1,1309 @@
+            ``colsample_bytree``, ``reg_alpha`` and ``reg_lambda`` are
+            searched.
+
+        refit:
+            If True, refit the estimator with the best found hyperparameters.
+
+        study:
+            Study corresponds to the optimization task. If None, a new study is
+            created.
+
+        timeout:
+            Time limit in seconds for the search of appropriate models. If
+            None, the study is executed without time limitation. If
+            ``n_trials`` is also set to None, the study continues to create
+            trials until it receives a termination signal such as Ctrl+C or
+            SIGTERM. This trades off runtime vs quality of the solution.
+
+        **kwargs:
+            Other parameters for the model. See
+            http:/lightgbm.readthedocs.io/en/latest/Parameters.html for more
+            parameters. Note, that **kwargs is not supported in sklearn, so it
+            may cause unexpected issues.
+
+    Attributes:
+        best_iteration_:
+            Number of iterations as selected by early stopping.
+
+        best_params_:
+            Parameters of the best trial in the ``Study``.
+
+        best_score_:
+            Mean cross-validated score of the best estimator.
+
+        booster_:
+            Trained booster.
+
+        n_features_:
+            Number of features of fitted model.
+
+        n_splits_:
+            Number of cross-validation splits.
+
+        objective_:
+            Concrete objective used while fitting this model.
+
+        study_:
+            Actual study.
+
+        refit_time_:
+            Time for refitting the best estimator. This is present only if
+            ``refit`` is set to True.
+
+    Examples:
+        >>> import optuna.integration.lightgbm as lgb
+        >>> from sklearn.datasets import load_boston
+        >>> reg = lgb.LGBMRegressor(random_state=0)
+        >>> X, y = load_boston(return_X_y=True)
+        >>> reg.fit(X, y)
+        LGBMRegressor(...)
+        >>> y_pred = reg.predict(X)
+    """
+
+    def predict(
+        self, X: TwoDimArrayLikeType, num_iteration: Optional[int] = None, **predict_params: Any
+    ) -> np.ndarray:
+        """Predict using the fitted model.
+
+        Args:
+            X:
+                Data.
+
+            num_iteration:
+                Limit number of iterations in the prediction. If None, if the
+                best iteration exists, it is used; otherwise, all trees are
+                used. If <=0, all trees are used (no limits).
+
+            **predict_params:
+                Always ignored. This parameter exists for compatibility.
+
+        Returns:
+            y_pred:
+                Predicted values.
+        """
+        X = check_X(X, accept_sparse=True, estimator=self, force_all_finite=False)
+
+        return self.booster_.predict(X, num_iteration=num_iteration)
+
+
+# alias classes
+OGBMClassifier = LGBMClassifier
+OGBMRegressor = LGBMRegressor
```

Comment:
For simplification, how about removing aliases? Without reading the source code, users have no way of knowing the difference between OGBMClassifier and LGBMClassifier.